File: app/src/main/java/org/dhis2/usescases/jira/JiraModule.java
package org.dhis2.usescases.jira;

import org.dhis2.data.dagger.PerFragment;

import dagger.Module;
import dagger.Provides;

/**
 * QUADRAM. Created by ppajuelo on 24/05/2018.
 */

@Module
public class JiraModule {

    @Provides
    @PerFragment
    JiraPresenter providesPresenter() {
        return new JiraPresenterImpl();
    }
}

File: app/src/main/java/org/dhis2/usescases/jira/JiraModule.kt
package org.dhis2.usescases.jira

import dagger.Module
import dagger.Provides
import org.dhis2.commons.di.dagger.PerFragment
import org.dhis2.commons.prefs.PreferenceProvider
import org.dhis2.commons.resources.ResourceManager
import org.dhis2.commons.schedulers.SchedulerProvider

@Module
class JiraModule {
    @Provides
    @PerFragment
    fun jiraViewModelFactory(
        preferenceProvider: PreferenceProvider,
        resourceManager: ResourceManager,
        schedulerProvider: SchedulerProvider,
    ): JiraViewModelFactory {
        return JiraViewModelFactory(preferenceProvider, resourceManager, schedulerProvider)
    }
}
